daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -293,102 +296,110 @@ class Engine(val config: EngineConfig = new EngineConfig(LanguageVersion.StableV
   private[engine] def interpretLoop(
       machine: Machine,
       time: Time.Timestamp,
-  ): Result[(SubmittedTransaction, Tx.Metadata)] = machine.withOnLedger("Daml Engine") { onLedger =>
-    var finished: Boolean = false
-    while (!finished) {
-      machine.run() match {
-        case SResultFinalValue(_) => finished = true
-
-        case SResultError(SError.DamlEDuplicateContractKey(key)) =>
-          // Special-cased because duplicate key errors
-          // produce a different gRPC error code.
-          return ResultError(Error.Interpretation(Error.Interpretation.DuplicateContractKey(key)))
-
-        case SResultError(SError.DamlEContractKeyNotFound(key)) =>
-          return ResultError(Error.Interpretation.ContractKeyNotFound(key))
-
-        case SResultError(err) =>
-          return ResultError(
-            Error.Interpretation.Generic(
-              s"Interpretation error: ${Pretty.prettyError(err, onLedger.ptxInternal).render(80)}",
-              s"Last location: ${Pretty.prettyLoc(machine.lastLocation).render(80)}, partial transaction: ${onLedger.ptxInternal.nodesToString}",
+  ): Result[(SubmittedTransaction, Tx.Metadata)] =
+    machine.withOnLedger(NameOf.qualifiedNameOfCurrentFunc) { onLedger =>
+      var finished: Boolean = false
+      while (!finished) {
+        machine.run() match {
+          case SResultFinalValue(_) => finished = true
+
+          case SResultError(err) =>
+            def details =
+              s"Last location: ${Pretty.prettyLoc(machine.lastLocation).render(80)}, partial transaction: ${onLedger.ptxInternal.nodesToString}"
+            val engineError: Error =
+              err match {
+                case SError.DamlEDuplicateContractKey(key) =>
+                  // Special-cased because duplicate key errors
+                  // produce a different gRPC error code.
```

Comment:
I think this comment has lost its meaning and can be deleted. Originally it made sense since not all errors where exposed via the `Engine` API. Now they are so this is no longer special.